backends/graphics/gles-custom/gles-custom.cpp

```diff
@@ -0,0 +1,703 @@
+
+		select(drm.fd+1, &fds, NULL, NULL, NULL);
+		
+		drmHandleEvent(drm.fd, &evctx);
+	}
+	
+	// release last buffer to render on again
+	gbm_surface_release_buffer(gbm.surface, bo);
+	bo = next_bo;
+}
+
+int init_drm(void)
+{
+	static const char *modules[] = {
+			"i915", "radeon", "nouveau", "vmwgfx", "omapdrm", "exynos", "msm"
+	};
+	drmModeRes *resources;
+	drmModeConnector *connector = NULL;
+	drmModeEncoder *encoder = NULL;
+	unsigned i, area;
+
+	// In plain C, we can just init evctx at declare time, but it's now allowed in C++
+	evctx.version = DRM_EVENT_CONTEXT_VERSION;
+	evctx.page_flip_handler = page_flip_handler;
+
+	for (i = 0; i < (sizeof(modules) / sizeof(modules[0])); i++) {
+		debug("trying to load module %s...", modules[i]);
+		drm.fd = drmOpen(modules[i], NULL);
+		if (drm.fd < 0) {
+			debug("failed.");
+		} else {
+			debug("success.");
+			break;
+		}
+	}
+
+	if (drm.fd < 0) {
+		debug("could not open drm device\n");
+		return -1;
+	}
+
+	resources = drmModeGetResources(drm.fd);
+	if (!resources) {
+		debug("drmModeGetResources failed\n");
+		return -1;
+	}
+
+	// find a connected connector
+	for (i = 0; i < (unsigned)resources->count_connectors; i++) {
+		connector = drmModeGetConnector(drm.fd, resources->connectors[i]);
+		if (connector->connection == DRM_MODE_CONNECTED) {
+			// it's connected, let's use this!
+			break;
+		}
+		drmModeFreeConnector(connector);
+		connector = NULL;
+	}
+
+	if (!connector) {
+		// we could be fancy and listen for hotplug events and wait for
+		// a connector..
+		debug("no connected connector found\n");
+		return -1;
+	}
+	// find highest resolution mode
+	for (i = 0, area = 0; i < (unsigned)connector->count_modes; i++) {
+		drmModeModeInfo *current_mode = &connector->modes[i];
+		unsigned current_area = current_mode->hdisplay * current_mode->vdisplay;
+		if (current_area > area) {
+			drm.mode = current_mode;
+			area = current_area;
+		}
+	}
+
+	if (!drm.mode) {
+		debug("could not find mode\n");
+		return -1;
+	}
+
+	// find encoder
+	for (i = 0; i < (unsigned)resources->count_encoders; i++) {
+		encoder = drmModeGetEncoder(drm.fd, resources->encoders[i]);
+		if (encoder->encoder_id == connector->encoder_id)
+			break;
+		drmModeFreeEncoder(encoder);
+		encoder = NULL;
+	}
+
+	if (!encoder) {
+		debug("no encoder found\n");
+		return -1;
+	}
+
+	drm.crtc_id = encoder->crtc_id;
+	drm.connector_id = connector->connector_id;
+
+	return 0;
+}
+
+int init_gbm(void)
+{
+	gbm.dev = gbm_create_device(drm.fd);
+
+	gbm.surface = gbm_surface_create(gbm.dev,
+			drm.mode->hdisplay, drm.mode->vdisplay,
+			GBM_FORMAT_XRGB8888,
+			GBM_BO_USE_SCANOUT | GBM_BO_USE_RENDERING);
+	if (!gbm.surface) {
+		debug ("failed to create gbm surface\n");
+		return -1;
+	}
+	
+	return 0;
+}
+
+#endif
+
+static struct {
+	EGLDisplay display;
+	EGLConfig config;
+	EGLContext context;
+	EGLSurface surface;
+	uint32_t width, height, refresh;
+} eglInfo;
+
+void OpenGLCustomGraphicsManager::init_egl(){
```

Comment:
Bracket formatting on this line, the following one and further down in similar cases.